client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/FlinkShuffleClientImpl.java

```diff
@@ -181,15 +181,11 @@ public CelebornBufferStream readBufferedPartition(
           shuffleId,
           partitionId,
           isSegmentGranularityVisible);
-      if (isSegmentGranularityVisible) {
-        // When the downstream reduce tasks start early than upstream map tasks, the shuffle
-        // partition locations may be found empty, should retry until the upstream task started
-        return CelebornBufferStream.empty();
-      } else {
-        throw new PartitionUnRetryAbleException(
-            String.format(
-                "Shuffle data lost for shuffle %d partition %d.", shuffleId, partitionId));
-      }
+      // TODO: in segment granularity visible senarios, when the downstream reduce tasks start early
```

Comment:
Is `isSegmentGranularityVisible` not used in current implementation?

Reply:
> Is `isSegmentGranularityVisible` not used in current implementation?

Yes, 'isSegmentGranularityVisible' is not used in the current plan, but retaining this field will not have a negative impact, and it can indicate on the server side whether the Flink integration strategy for the current job is hybrid shuffle.